{{cookiecutter.github_repository}}/{{cookiecutter.main_module}}/base/utils/urls.py

```diff
@@ -41,3 +42,9 @@ def reverse(viewname, *args, **kwargs):
     """Same behavior as django's reverse but uses django_sites to compute absolute url.
     """
     return get_absolute_url(django_reverse(viewname, *args, **kwargs))
+
+
+def build_urlencoded_query_params(query_params_dict):
```

Comment:
I would prefer sending in the url as well along with the query_params_dict. The function can then return the full url with url encoded params.

Reply:
I'm curious how it is different from [`urlecode`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode)? Which has `quote_via=quote_plus` as default parameter.

> urllib.parse.urlencode(query, doseq=False, safe='', encoding=None, errors=None, quote_via=quote_plus)

> The resulting string is a series of key=value pairs separated by '&' characters, where both key and value are quoted using the quote_via function. By default, quote_plus() is used to quote the values, which means spaces are quoted as a '+' character and ‘/’ characters are encoded as %2F, which follows the standard for GET requests (application/x-www-form-urlencoded). An alternate function that can be passed as quote_via is quote(), which will encode spaces as %20 and not encode ‘/’ characters. For maximum control of what is quoted, use quote and specify a value for safe.

Here I only see `value` being encode and not the `key`.

Reply:
Here is the source code for `urlencode` which I don't see much different than what is being done here: https://github.com/Fueled/django-init/pull/409#pullrequestreview-442191808

Reply:
Okay, I think the only difference is that urlencode accepts the only dict as DS. While in our implementation we can have list of tuples that can help in preserving the order of query params (which we aren't currently doing).
Also, I don't think it would be particularly helpful to be included in Django-init when we can directly use urlencode.

I'll close this PR.

Thanks for your response :)

Reply:
cool.